### DISCUSSION TITLE
Why wrapping_add is used for integer addition?

### DISCUSSION THREAD

**Discussion by hnakamur** [mitsuhiko/minijinja#348] [Q&A] [answered]
First of all, thanks for creating and sharing nice template engines such as Jinja2 and MiniJinja!

Why wrapping_add is used for integer addition?
https://github.com/mitsuhiko/minijinja/blob/edbaf5924cef2ed4bdee319d510e217b6207441c/minijinja/src/value/ops.rs#L183

I noticed checked version of operation like checked_sub or checked_mul are used for other operations.
Why integer addition is different?

One of the reason I am asking this is since I am porting MiniJinja to Go and I would like to it to be compatible as MiniJinja as possible.
Currently I implemented wrappring_add for i128 using Go's math/big.Int.

https://github.com/hnakamur/mjingo/blob/b36cb7d5788c5aab21024a88bccd4f35c4271e4c/ops.go#L366-L377

But I am wondering is it really correct or maybe checked_add should be used instead.

**Comment by mitsuhiko** [comment, marked as answer]
An oversight. I will change this to checked which is more in line with the behavior of other operations.

Changing in #350

**Reply by hnakamur** [reply]
Thanks!
